src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -110,7 +110,7 @@ internal partial class TagComputer
             public event EventHandler<SnapshotSpanEventArgs>? TagsChanged;
 
             private IClassificationService? TryGetClassificationService(ITextSnapshot snapshot)
-                => _workspace?.Services.GetLanguageServices(snapshot.ContentType)?.GetService<IClassificationService>();
+                => _workspace?.Services.SolutionServices.GetProjectServices(snapshot.ContentType)?.GetService<IClassificationService>();
```

Comment:
Is this not coming from a document because we still try to do syntax classification even if we have no document at all?

Reply:
you know what.  this is pretty darn suspect code :)  i can't imagine why we would have this (Except that this is insanely old code).

basically we bail out a line below if we don't have a documetn anyways... so we should always expect to have a Doc + classification service.  I will remove in a separate PR.